### ISSUE TITLE
Export still shows code

### ISSUE THREAD

**Issue by jkleint** [kirbs-/hide_code#23]
Very nicely done extension, thank you.  Works great when viewing the notebook in Jupyter, but doesn't seem to work on export with the latest version of Jupyter.

First, it doesn't install; I think the location for extensions has moved to `~/.local/share/jupyter/nbextensions/`.  I installed it manually.

The toolbar buttons show up in the viewer, and clicking the toggle button hides code; but all of the export buttons redirect to the wrong URL.  They're trying to go to `/notebooks/mynotebook.ipynb/export/html`, but it looks like the new location is `/nbconvert/html/mynotebook.ipynb?download=true`.  At least that's what I get when I use the File -> Download As menu.

Speaking of which, when I do Download As HTML, the inputs aren't hidden.

OK, I'll try the nbconvert command line.  Same problem; no hidden inputs.  Only if I View -> Cell Toolbar -> Hide Code and then manually check 'Hide Code' for each cell and then save the notebook does it work, and then only on the command line (not the in-viewer export).

Bonus issue: when I try `jupyter-nbconvert --to hide_code_latexpdf`, I get `ImportError: No module named HideCodeLatexPDFExporter` (which does seem to be nonexistent).

Here's what I'm running.
```
Python 2.7.10 (default, Oct 14 2015, 16:09:02) 
[GCC 5.2.1 20151010] on linux2

hide-code==0.3.0
ipykernel==4.3.1
ipython==5.0.0
ipython-genutils==0.1.0
ipywidgets==5.2.2
jupyter==1.0.0
jupyter-client==4.3.0
jupyter-console==5.0.0
jupyter-core==4.1.0
nbconvert==4.2.0
nbformat==4.0.1
notebook==4.2.2
widgetsnbextension==1.2.6
```

Seems like Jupyter is a fast moving target these days.  I appreciate you trying to keep up!

**Comment by kirbs-** [comment]
5.0 just came out a couple of weeks ago. I haven't tested against that yet. Would you post the output from the pip install?

Also, the exporting issues you mention should go away if you restart Juypter. The import errors you're seeing usually happen when hide_code is installed with Juypter already running. Checkout issue #21.

**Comment by jkleint** [comment]
Thanks for the response.  The pip output is not interesting, except that it does not contain any of the `Running setup.py install` or post-install stuff, which would explain why it's not getting installed.

I also discovered that a pip uninstall does not remove the extension, since the JS (should) get copied into `~/.local/share/jupyter/nbextensions/`, and pip does not remove things from there.  Perhaps there's some way to hook into whatever `jupyter nbextension uninstall` is doing.

The exporting issues are still issues after restart; I think the URLs have just changed.

I made sure Jupyter wasn't running when installing hide_code, but still get the import error.  The string `HideCodeLatexPDFExporter` is not physically present in any Python file.

Also, and this is actually a problem, once you've done 'hide code' on the cell toolbar, then save and re-open the notebook, the 'Hide/show code' button does not work, and your code is gone with no easy way to retrieve it.  That's non-awesome.

Thanks again.

**Comment by kirbs-** [comment]
I think the problem you're seeing is due to a manual install into the nbextensions directory. The JS for the UI is being loaded, but all the backend stuff isn't.

Can you paste the output from Jupyter's console right after startup?  It should look like below.

```
[W 19:02:32.053 NotebookApp] Unrecognized JSON config file version, assuming version 1
[I 19:02:37.352 NotebookApp] hide_code: Attempting to load hid_code export handler extensions.
[I 19:02:37.354 NotebookApp] hide_code: Hide_code export handler extensions loaded.
[I 19:02:37.355 NotebookApp] Serving notebooks from local directory: /Users/****/git/esp8266-micropython-vagrant/ESP8266
[I 19:02:37.355 NotebookApp] 0 active kernels 
[I 19:02:37.355 NotebookApp] The Jupyter Notebook is running at: http://localhost:8888/
[I 19:02:37.355 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```
Notice the second and third lines. This indicates the backend server extensions are loaded.

**Comment by JRocki** [comment]
I am having similar issues.  Really do like this though!

**Comment by kirbs-** [comment]
Can you paste the output from Jupyter's console right after startup? It should look like below.
```
[W 19:02:32.053 NotebookApp] Unrecognized JSON config file version, assuming version 1
[I 19:02:37.352 NotebookApp] hide_code: Attempting to load hid_code export handler extensions.
[I 19:02:37.354 NotebookApp] hide_code: Hide_code export handler extensions loaded.
[I 19:02:37.355 NotebookApp] Serving notebooks from local directory: /Users/****/git/esp8266-micropython-vagrant/ESP8266
[I 19:02:37.355 NotebookApp] 0 active kernels 
[I 19:02:37.355 NotebookApp] The Jupyter Notebook is running at: http://localhost:8888/
[I 19:02:37.355 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

**Comment by lum4chi** [comment]
Don't know if this can help: I experienced some trouble too loading this extension. I've been able to load this by forcing it in jupyter_notebook_config.json. It should appear something like:
{
  "NotebookApp": {
    "nbserver_extensions": {
      "jupyter_nbextensions_configurator": true,
      "hide_code.hide_code": true
    },
    "extra_template_paths": [
      "/Users/XXX/Library/Jupyter/templates"
    ]
  },
  "version": 1
}
But now it works like a charm!

Hope it helps: by the way I'm able to confirm that all is running with package updated at the time that I'm writing.

**Comment by kirbs-** [comment]
Thanks! Are you able to export notebooks with code hidden too?

**Comment by lum4chi** [comment]
Yes, but only when correct used cell checkboxes (<\> work only as preview) and, as required, notebook needs to be saved in order to export. PDF export works too, but page format is huge (code is small compared to page). Maybe is fault of an unknown configuration. Amy suggestion?

**Comment by kirbs-** [comment]
I'll test this out. I've been meaning to support nbextension installation.
